presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -979,7 +979,16 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
                     throw new SemanticException(INVALID_PROCEDURE_ARGUMENTS, initialValueArg, "REDUCE_AGG only supports non-NULL literal as the initial value", initialValueArg);
                 }
             }
+
             Type type = functionAndTypeManager.getType(functionMetadata.getReturnType());
+
+            if (StandardTypes.MAP.equals(functionMetadata.getReturnType().getBase())) {
+                String keyTypeBase = functionMetadata.getReturnType().getTypeParametersAsTypeSignatures().get(0).getBase();
```

Comment:
Is there a possibility this can hit null?

Reply:
Map always have two parameters - key, value. So, it should not have null. I saw similar assumption in existing codebase as well